plotly/figure_factory/_facet_grid.py

```diff
@@ -0,0 +1,1073 @@
+                                  facet_row)
+            annotations.append(
+                _annotation_dict(label, num_of_rows - row_count, num_of_rows,
+                                 row_col='row', flipped=flipped_rows)
+                )
+
+    # add annotations
+    fig['layout']['annotations'] = annotations
+
+    return fig
+
+
+def create_facet_grid(df, x, y, facet_row=None, facet_col=None,
+                      color_name=None, colormap=None, color_is_cat=False,
+                      widen_frame=False, facet_row_labels=None,
+                      facet_col_labels=None, height=700, width=700, size=5,
+                      trace_type='scatter', scales='fixed', dtick_x=None,
+                      dtick_y=None, show_boxes=True, **kwargs):
+    """
+    Returns figure for facet grid.
+
+    :param (pd.DataFrame) df: the dataframe of columns for the facet grid.
+    :param (str) x: the key of the dataframe to be used as the x axis df.
+    :param (str) y: the key of the dataframe to be used as the y axis df.
+    :param (str) facet_row: the key for row filter column for the facet grid.
+    :param (str) facet_col: the key for the column filter column for the facet
+        grid.
+    :param (str) color_name: the name of your dataframe column that will
+        function as the colormap variable.
+    :param (str|list|dict) colormap: the param that determines how the
+        color_name column colors the data. If the dataframe contains numeric
+        data, then a dictionary of colors will group the data categorically
+        while a Plotly Colorscale name or a custom colorscale will treat it
+        numerically. To learn more about colors and types of colormap, run
+        `help(plotly.colors)`.
+    :param (bool) color_is_cat: determines whether a numerical column for the
+        colormap will be treated as categorical (True) or sequential (False).
+            Default = False.
+    :param (bool) widen_frame: if set to True, all points in each subplot
+        are strickly contained in the region of the subplot by increasing the
+        maximum and minimum range values by 1. Setting to False doesn't do
+        anything.
+        Default = False
+    :param (str|dict) facet_row_labels: set to either 'name' or a dictionary
+        of all the values in the facetting row mapped to some text to show up
+        in the label annotations. If None, labelling works like usual.
+    :param (str|dict) facet_col_labels: set to either 'name' or a dictionary
+        of all the values in the facetting row mapped to some text to show up
+        in the label annotations. If None, labelling works like usual.
+    :param (int) height: the height of the facet grid figure.
+    :param (int) width: the width of the facet grid figure.
+    :param (int) size: point size in pixels.
+    :param (str) trace_type: decides the type of plot to appear in the
+        facet grid. The options are 'scatter' and 'scattergl'.
```

Comment:
In order for histogram to work, you'd need to also make `x` and `y` optional too